src/ucai/core/utils/callable_utils.py

```diff
@@ -0,0 +1,142 @@
+import ast
+import inspect
+from textwrap import dedent
+from typing import Any, Callable, get_type_hints
+
+from ucai.core.utils.type_utils import python_type_to_sql_type, validate_container_type
+
+FORBIDDEN_PARAMS = ['self', 'cls']
+
+def extract_function_body(func: Callable[..., Any]) -> str:
+    """Extracts the body of a function as a string without the signature or docstring, preserving indentation."""
```

Comment:
Do we need to preserve indentation?

Reply:
Can we just `textwrap.indent` the body when injecting it in `CREATE FUNCTION`?

Reply:
Great point. Refactored the implementation to adjust for 2 vs 4 indentation and simplified this dedent/indent logic. Validated in DBR that the indentation works for function execution